src/explore-education-statistics-common/src/modules/table-tool/utils/__tests__/optimizeFilters.test.ts

```diff
@@ -0,0 +1,41 @@
+import optimizeFilters from '../optimizeFilters';
+import {
+  testFilters1,
+  testHeaderConfig1,
+  testCategoryFilter1,
+  testFilters2,
+  testHeaderConfig2,
+  testCategoryFilter4,
+  testFilters3,
+  testHeaderConfig3,
+  testCategoryFilter8,
+  testFilters4,
+  testHeaderConfig4,
+  testLocationFilter1,
+} from '../__data__/testFilterData';
+
+describe('optimizeFilters', () => {
+  test('reduces three identical category filters down to one', () => {
+    expect(optimizeFilters(testFilters1, testHeaderConfig1)).toEqual([
+      testCategoryFilter1,
+    ]);
+  });
+
+  test('adds additional filter subgroups if required', () => {
+    expect(optimizeFilters(testFilters2, testHeaderConfig2)).toEqual([
+      testCategoryFilter4,
+    ]);
+  });
```

Comment:
Similar to above, this test case doesn't seem right. Contrary to what it says, we don't end up with an extra subgroup in the result.